src/urllib3/__init__.py

```diff
@@ -202,3 +203,9 @@ def request(
         timeout=timeout,
         json=json,
     )
+
+
+if sys.platform == "emscripten":
+    from .contrib.emscripten import inject_into_urllib3  # noqa: 401
+
+    inject_into_urllib3()
```

Comment:
Have we considered *not* doing this by default?

Reply:
Yeah that seems correct, we can document that you need to call this for Emscripten support.

Reply:
Personally I really think it makes more sense to make it work out of the box on emscripten. Otherwise you end up with the situation that someone downloads a library which depends on requests or urllib3 and it inexplicably just doesn't work at all, and the only way they can make it work is to know that they need to read the dependency list, find urllib3 on it, and then read the urllib3 documentation to find that while urllib3 does support emscripten, you need to call a special function before urllib3 works on emscripten, and without it you just get a range of obscure not implemented errors inside socket We're already seeing similar with the current pyodide-http monkeypatch module, where you end up putting in code downstream for emscripten support that basically just looks for sys.platform and calls pyodide-http to monkeypatch everything.

Reply:
One possible thing to do would be to add a check for whether sockets are implemented into this code just in case something changes in the range of environments targeted by emscripten. Realistically though, sockets support is really unlikely in browser webasembly, so it is likely we'll remain in the situation where only http based protocols are available for at least the foreseeable future.

Reply:
I don't have a strong opinion, just thought it should be considered.

What are the arguments in favor of opting in? Explicit is better than implicit, mostly. Having urllib3 work magically could give the impression that sockets *are* supported and set unrealistic expectations. More hypothetically, having an explicit API gives us more options.

Your arguments against opting in do make sense though!

Reply:
We can stick to this for now, I want to poke around for a bit once this is merged too.